pymongo/client_session.py

```diff
@@ -979,16 +988,20 @@ def _apply_to(
         read_preference: _ServerMode,
         conn: Connection,
     ) -> None:
+        if not conn.supports_sessions:
+            if not self._implicit:
+                raise ConfigurationError("Sessions are not supported by this MongoDB deployment")
+            return
         self._check_ended()
-        self._materialize()
+        self._materialize(conn.logical_session_timeout_minutes)
         if self.options.snapshot:
             self._update_read_concern(command, conn)
 
         self._server_session.last_use = time.monotonic()
         command["lsid"] = self._server_session.session_id
 
         if is_retryable:
-            command["txnNumber"] = self._server_session.transaction_id
+            command["txnNumber"] = self._transaction_id
```

Comment:
I'd say change these back to _server_session.transaction_id since we've already called materialize().